File: supermarket-app/src/app/models/product.ts
export interface ProductCategory {
  id: number;
  name: string;        
  description: string; 
  icon: string;        
  displayOrder: number;
}

export interface Product {
  id: number;
  snowflakeId: string;
  name: string;
  price: number;
  description?: string;
  stockQuantity: number;
  categoryId: number;
  category: ProductCategory;
  photo: string;
  originalPrice?: number;
  brand?: string;
  weight?: number;
  unit?: string;
  isOnSale?: boolean;
  promotionNames?: string[];

}

export interface ProductDto{
  id: number;
  snowflakeId: string;
  name: string;
  price: number;
  photo: string;
  isOnSale: boolean;
  originalPrice?: number;
  promotionNames?: string[];
}

export interface PagedResult<T> {
  items: T[];
  totalCount: number;
  pageNumber: number;
  pageSize: number;
  totalPages: number;
}

File: SupermarketMock/SupermarketMock/Models/Product.cs

    namespace SupermarketMock.Models
    {

        public class Product
        {
            public int Id { get; set; }

            public long SnowflakeId { get; set; }

            public string Name { get; set; } = string.Empty;

            public string? Description { get; set; }

            public decimal Price { get; set; }

            public int CategoryId { get; set; }                    
            public ProductCategory Category { get; set; } = null!;    

            public string? SubCategory { get; set; }

            public string Photo { get; set; } = string.Empty;

            public int StockQuantity { get; set; }

            public bool IsAvailable { get; set; } = true;

            public DateTime? ExpiryDate { get; set; }

            public string? Brand { get; set; }

            public decimal? Weight { get; set; }
            public string? Unit { get; set; } = "piece";

            public int? Rating { get; set; }
            public int? ReviewCount { get; set; }
            public ICollection<ProductPromotion> ProductPromotions { get; set; } = new List<ProductPromotion>();

        }
        



    }

